mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -605,6 +619,90 @@ public void onClick(View view) {
+        }
+
+        @Override
+        public View onCreateView(@NonNull LayoutInflater inflater,
+                ViewGroup container, Bundle savedInstanceState) {
+            View view = inflater.inflate(R.layout.fragment_habpanel, container, false);
+
+            TextView retryButton = view.findViewById(R.id.retry_button);
+            retryButton.setOnClickListener(v -> loadHabpanel(view));
+            loadHabpanel(view);
+
+            return view;
+        }
+
+        private void loadHabpanel(View view) {
+            updateViewVisibility(false, true, view);
+            try {
+                mConnection = ConnectionFactory.getUsableConnection();
+            } catch (ConnectionException e) {
+                updateViewVisibility(true, false, view);
+                return;
+            }
+
+            if (mConnection == null) {
+                updateViewVisibility(true, false, view);
+                return;
+            } else {
+                updateViewVisibility(false, true, view);
+            }
+
+            HttpUrl url = mConnection.getAsyncHttpClient().buildUrl("/habpanel/index.html");
+
+            WebView webView = view.findViewById(R.id.webview);
+
+            webView.setWebViewClient(new AnchorWebViewClient(url.toString(),
+                    mConnection.getUsername(), mConnection.getPassword()) {
+                @Override
+                public void onPageFinished(WebView view, String url) {
+                    updateViewVisibility(false, false, getView());
+                }
+
+                @Override
+                public void onReceivedError(WebView view, WebResourceRequest request,
+                        WebResourceError error) {
+                    updateViewVisibility(true, false, getView());
+                }
+
+                @Override
+                public void onReceivedHttpError(WebView view, WebResourceRequest request,
+                        WebResourceResponse errorResponse) {
+                    updateViewVisibility(true, false, getView());
+                }
+            });
+            webView.getSettings().setDomStorageEnabled(true);
+            webView.getSettings().setJavaScriptEnabled(true);
+            webView.loadUrl(url.toString());
+        }
+
+        private void updateViewVisibility(boolean error, boolean loading, @NonNull View view) {
+            if (view == null) {
+                return;
+            }
+            view.findViewById(R.id.webview).setVisibility(error ? View.GONE : View.VISIBLE);
+            view.findViewById(android.R.id.empty).setVisibility(error ? View.VISIBLE : View.GONE);
+            view.findViewById(R.id.progress).setVisibility(loading ? View.VISIBLE : View.GONE);
+        }
+
+        @Override
+        public void onAvailableConnectionChanged() {
+            loadHabpanel(getView());
+        }
+
+        @Override
+        public void onCloudConnectionChanged(CloudConnection connection) {
+            loadHabpanel(getView());
```

Comment:
Why this? Habpanel doesn't care about the cloud connection...

Reply:
HABPanel should only be refresh on onAvailableConnectionChanged.